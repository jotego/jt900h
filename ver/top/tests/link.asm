    ; RRD
    main section code
    org 0
    ld a,0xbf    ; common header

    ld xsp,stack
    ld xwa,0x1234678
    link xwa,16
    cp xsp,stack-4-16
    jp ne,bad_end
    cp xwa,stack-4
    jp ne,bad_end

    ld xbc,xsp
    add xsp,16
    pop xde
    cp xde,0x1234678
    jp ne,bad_end

test_end:
    ; ld (0xffff),0xff
end_loop:
    ldf 0
    ld hl,0xbabe
    jp end_loop
bad_end:
    ld hl,0xdead
    jp bad_end
data:
    db 0x12,0x34,0x56,0x78,0x9A,0xBC,0xDE,0xF0
    dw 0xbbbb,0x0000,0x0000,0x0000
ref:
    db 0x01,0x23,0x45,0x67,0x89,0xAB,0xCD,0xEF
    dw 0x0000,0x0000,0x0000,0x0000
    dw 0x0000,0x0000,0x0000,0x0000,0x0000,0x0000,0x0000,0x0000
copy:
    dw 0x0000,0x0000,0x0000,0x0000,0x0000,0x0000,0x0000,0x0000
stack:
    end