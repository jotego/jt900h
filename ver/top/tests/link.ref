00000090 - 00000000 - 00000000 - 00000000
00000080 - 00000000 - 00000000 - 00000000
01234678 - 00000000 - 00000000 - 00000000
0000babe - 00000000 - 00000000 - 00000000
-- index registers --
00000000 - 00000000 - 00000000 - 00000094
SR = f842
