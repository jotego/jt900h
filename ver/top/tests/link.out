00000086 - 00000000 - 00000000 - 00000000
00000076 - 00000000 - 00000000 - 00000000
01234678 - 00000000 - 00000000 - 00000000
0000babe - 00000000 - 00000000 - 00000000
-- index registers --
00000000 - 00000000 - 00000000 - 0000008a
SR = 42f8
